examples/seismic/model.py

```diff
@@ -322,7 +372,71 @@ def initialize_function(function, data, nbpml):
     function.data_with_halo[:] = np.pad(data, pad_list, 'edge')
 
 
-class Model(object):
+class Pysical_Model(object):
+    """
+    General model class with comon properties
+    """
+    def __init__(self, origin, spacing, shape, space_order, nbpml=40,
+                 dtype=np.float32):
+        self.shape = shape
+        self.nbpml = int(nbpml)
+        self.origin = tuple([dtype(o) for o in origin])
+
+        shape_pml = np.array(shape) + 2 * self.nbpml
+        # Physical extent is calculated per cell, so shape - 1
+        extent = tuple(np.array(spacing) * (shape_pml - 1))
+        self.grid = Grid(extent=extent, shape=shape_pml,
+                         origin=origin, dtype=dtype)
+
+    def physical_params(self, **kwargs):
+        """
+        Return all set physical parameters and update to input values if provided
+        """
+        known = [getattr(self, i) for i in self._physical_parameters]
+        return {i.name: kwargs.get(i.name, i) or i for i in known}
+
+    @property
+    def dim(self):
+        """
+        Spatial dimension of the problem and model domain.
+        """
+        return self.grid.dim
+
+    @property
+    def spacing(self):
+        """
+        Grid spacing for all fields in the physical model.
+        """
+        return self.grid.spacing
+
+    @property
+    def spacing_map(self):
+        """
+        Map between spacing symbols and their values for each :class:`SpaceDimension`
+        """
+        return self.grid.spacing_map
+
+    @property
+    def dtype(self):
+        """
+        Data type for all assocaited data objects.
+        """
+        return self.grid.dtype
+
+    @property
+    def shape_domain(self):
+        """Computational shape of the model domain, with PML layers"""
+        return tuple(d + 2*self.nbpml for d in self.shape)
+
+    @property
+    def domain_size(self):
+        """
+        Physical size of the domain as determined by shape and spacing
+        """
+        return tuple((d-1) * s for d, s in zip(self.shape, self.spacing))
+
+
+class Model(Pysical_Model):
```

Comment:
some parameters appear in the docstring, but not in the model (probably leftovers). I spotted `rho`, for instance

Reply:
reminder